fiaas_skipper/deploy/deploy.py

```diff
@@ -32,20 +32,23 @@ def __init__(self, cluster, release_channel_factory, bootstrap, spec_config=None
             self._spec_config = spec_config
 
     def deploy(self, namespaces=None):
-        deploy_counter.inc()
-        last_deploy_gauge.set_to_current_time()
-        deployment_configs = self._cluster.find_deployment_configs(NAME)
-        fiaas_enabled_namespaces_gauge.set(len(deployment_configs))
-        for deployment_config in deployment_configs:
-            if namespaces and deployment_config.namespace not in namespaces:
-                continue
-            channel = self._release_channel_factory(deployment_config.name, deployment_config.tag)
-            try:
-                self._deploy(deployment_config, channel)
-                if requires_bootstrap(deployment_config):
-                    self._bootstrap(deployment_config, channel, self._spec_config)
-            except Exception:
-                LOG.exception("Failed to deploy %s in %s", deployment_config.name, deployment_config.namespace)
+        try:
+            deploy_counter.inc()
+            last_deploy_gauge.set_to_current_time()
+            deployment_configs = self._cluster.find_deployment_configs(NAME)
+            fiaas_enabled_namespaces_gauge.set(len(deployment_configs))
+            for deployment_config in deployment_configs:
+                if namespaces and deployment_config.namespace not in namespaces:
+                    continue
+                channel = self._release_channel_factory(deployment_config.name, deployment_config.tag)
+                try:
+                    self._deploy(deployment_config, channel)
+                    if requires_bootstrap(deployment_config):
+                        self._bootstrap(deployment_config, channel, self._spec_config)
+                except Exception:
+                    LOG.exception("Failed to deploy %s in %s", deployment_config.name, deployment_config.namespace)
+        except Exception:
+            LOG.exception("Unexpected failure when deploying")
```

Comment:
This is functionally fine the way things work right now. However, the reason we're adding this catch-all error handling is because we are using this callable to spawn a new thread. If this method is called from elsewhere at a later point, it might not make sense that all exceptions are caught here and not propagated. To avoid this we could wrap the callable in a function that only does the try/catch block and calls its input at the thread creation site. Does that make sense?

Reply:
Yes, I've moved the catch-all to the web module now.